tests/sources/test_network_drive.py

```diff
@@ -276,82 +275,68 @@ async def test_fetch_file_when_file_is_inaccessible(file_mock):
         file_mock.side_effect = SMBOSError(ntstatus=0xC0000043, filename="file1.txt")
 
         # Execute
-        response = source.fetch_file_content(path=path)
+        async for response in source.fetch_file_content(path=path):
+            # Assert
+            assert response is None
```

Comment:
I know it wasn't part of your PR specifically, but this test really should assert more than `response is None`. Can you add a check to make sure the expected logger is output?